jax_md/quantity.py

```diff
@@ -459,6 +459,56 @@ def g_fn(R, neighbor):
       return g_R
   return neighbor_fn, g_fn
 
+def nball_unit_volume(spatial_dimension: int) -> float:
+  return jnp.power(jnp.pi, spatial_dimension / 2) / \
+    jnp.exp( jsp.special.gammaln(spatial_dimension / 2 + 1))
+
+def particle_volume(radii: Array, 
```

Comment:
If the user supplies species, would we expect per-particle radii or per-species? I sort of naively would expect the latter (or for something that mimics the convention of `smap` in this regard). But just glancing at the code (and comments) it always seems like this function requires a flat array of length num atoms?

Reply:
The only time radii is should have size num atoms is in the "case 2" example above. In cases 3-5 (including whenever species is specified) radii is interpreted as per-species. I think the confusion was in the documentation for radii, which says it should have shape `(n,)`... but this `n` doesn't have to be the number of particles. I changed the documentation to hopefully make this more clear. Perhaps I should also add the 5 examples?